api/api/validators/generate_teams_validator.py

```diff
@@ -7,23 +14,59 @@ def validate(self):
         self.validate_algorithm_type()
         self.validate_team_options()
         self.validate_algorithm_options()
-        self.validate_projects()
+        self.validate_algorithm_configs()
         self.validate_students()
 
     def _validate_schema(self):
-        pass
+        Schema(dict).validate(self.data)
+        Schema(list).validate(self.data.get("students"))
+        Schema(dict).validate(self.data.get("algorithm_options"))
+        Schema(dict).validate(self.data.get("team_generation_options"))
 
     def validate_algorithm_type(self):
-        pass
+        algorithm_type = self.data.get("algorithm_options").get("algorithm_type")
+        Schema(str).validate(algorithm_type)
+        try:
+            AlgorithmType(algorithm_type)
+        except ValueError:
+            raise SchemaError(
+                f"Algorithm type {algorithm_type} is not a valid algorithm type."
+            )
 
     def validate_algorithm_options(self):
-        pass
+        algorithm_option = dict.copy(self.data.get("algorithm_options"))
+        algorithm_type = AlgorithmType(algorithm_option.pop("algorithm_type"))
+        algorithm_option_cls = AlgorithmRunner.get_algorithm_option_class(
+            algorithm_type
+        )
+
+        algorithm_option_cls(**algorithm_option).validate()
```

Comment:
we shouldn't really be introducing data class inits at this level imo, this class is just a screen of the raw data to make sure it is able to be converted into the dataclasses properly by the loader. 

When the loader creates these objects it will run the validate() on the dataclass automatically so replicating that logic here doesn't grant us anything

Reply:
this applies to the other methods you have here